examples/simple/main.tf

```diff
@@ -14,47 +14,36 @@
  * limitations under the License.
  */
 
-variable region {
-  default = "us-central1"
-}
-
-variable network {
-  default = "default"
-}
-
-variable zone {
-  default = "us-central1-b"
-}
-
-provider google {
-  region = "${var.region}"
+provider "google" {
+  region = var.region
 }
 
 module "gce-lb-fr" {
   source       = "GoogleCloudPlatform/lb/google"
-  version      = "1.0.2"
-  region       = "${var.region}"
-  network      = "${var.network}"
+  version      = "~> 2.0"
+  region       = var.region
+  network      = var.network
+  project      = var.project
   name         = "group1-lb"
-  service_port = "${module.mig1.service_port}"
-  target_tags  = ["${module.mig1.target_tags}"]
+  service_port = var.service_port
```

Comment:
I don't see why we should hard code the `named_ports` instead of making it a variable, like it's done for instance here https://github.com/terraform-google-modules/terraform-google-vm/blob/70949e5dcc70a5d06baf04f4aefc9704420656b1/examples/mig/full/variables.tf#L54-L61. Also the following approach won't work:
```hcl
module "mig1" {
  source             = "terraform-google-modules/vm/google//modules/mig"
  project_id         = var.project_id
  subnetwork_project = var.subnetwork_project
  version            = "~> 1.1"
  region             = var.region
  target_pools       = [module.gce-lb-fr.target_pool]
  instance_template  = module.instance_template1.self_link
  hostname           = "mig1"
  named_ports        = [{
    name = "http"
    port = 80
  }]

...
module "gce-lb-fr" {
  source       = "GoogleCloudPlatform/lb/google"
  version      = "~> 2.0"
  region       = var.region
  network      = var.network
  project      = var.project
  name         = "group1-lb"
  service_port = module.mig1.named_ports[0].port
  ...
}  
```
because it should be an output in order to make a reference:
```
Error: Reference to undeclared output value

  on main.tf line 28, in module "gce-lb-fr":
  28:   service_port = module.mig1.named_ports[0].port

An output value with the name "named_ports" has not been declared in
module.mig1.
```

Reply:
Fixed